File: modules/core/src/main/java/fr/rozanc/cocozzelle/datatable/JacksonObjectViewFactory.java
package fr.rozanc.cocozzelle.datatable;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.List;
import java.util.Map;

public class JacksonObjectViewFactory implements ObjectViewFactory {

    private final ObjectMapper objectMapper;

    JacksonObjectViewFactory(final ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }

    @Override
    public DataTableView createView(final Object object, final List<ColumnExpression> columnExpressions) {
        final Map<String, Object> objectAsMap = objectMapper.convertValue(object, Map.class);
        return null;
    }
}

File: modules/core/src/main/kotlin/fr/rozanc/kokozzelle/datatable/JacksonObjectViewFactory.kt
package fr.rozanc.kokozzelle.datatable

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.convertValue
import fr.rozanc.kokozzelle.dataview.DataTableRowView

class JacksonObjectViewFactory internal constructor(private val objectMapper: ObjectMapper) : ObjectViewFactory {

    override fun createView(obj: Any?, columnExpressions: List<ColumnExpression>): DataTableRowView {
        val objectAsMap: Map<String, Any?> = obj?.let { objectMapper.convertValue(it) } ?: emptyMap()
        return DataTableRowView(objectAsMap, )
    }
}
